### PR TITLE
Add 0.11 news

### PR DESCRIPTION
**Pull request by peternewman**
All PRs as handy links. I'd suggest initially people do suggested changes to turn them into a brief summary of each change (which may be multiple within some PRs) and then we can work on categorising them afterwards. We generally only put a #num to refer to an issue if there's one it's closed or made progress on. I've added a few of those after the dash where they were obvious.

 Uncategorised and unsummarised PRs:
#1000
#1008
#1014
#1015
#1035
#1039
#1045
#1052
#1067
#1080 - #1028
#1100
#1134
#1136
#1138
#1140
#1142
#1146
#1147
#1176
#1181
#1185
#1196
#1209
#1213
#1215
#1231
#1237
#1242
#1260
#1268
#1269
#1270
#1273
#1275
#1277
#1289
#1292
#1294
#1295
#1297
#1298
#1299
#1300
#1301 - #1290
#1303
#1304
#1305
#1307
#1310
#1312
#1314
#1319
#1326
#1329
#1336
#1342
#1343
#1348
#1350
#1355
#1358
#1359
#1369
#1380
#1389
#1391
#1405
#1410
#1415
#1417
#1450
#1457
#1467
#1476
#1490
#1494
#1512
#1520
#1522 - Towards #1396
#1525
#1532
#1535
#1556
#1561
#1608 - #1592
#1620
#1621
#1625
#1633
#1647
#1648
#1683
#1701
#1705
#1706
#1712
#1717
#1720
#1787 - #1544
#1807
#1826
#1828
#1830
#1831
#1832
#1834
#1840
#1854
#1855
#1888
#1896
#1898
#1901
#1904
#1914
#1926
#1932
#1953
#1976
#1978
#1981
#1983
#1986
#1996
#1999
#2004
#2014
#2032
#2041
#2042
#2046
#2052
#2054

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,3 +1,143 @@
...
+#1981
+#1983
+#1986
+#1996
+#1999
+#2004
+#2014
+#2032
+#2041
+#2042
```
Comment (peternewman) [inline NEWS:132]:
```suggestion
* Add support for NORTLE FTDI interfaces #1659
```

File: NEWS [hunk 2]
```diff
@@ -1,3 +1,143 @@
...
+#1904
+#1914
+#1926
+#1932
+#1953
+#1976
+#1978
+#1981
+#1983
+#1986
```
Comment (peternewman) [inline NEWS:125]:
```suggestion
 * Fix some minor typos
```

File: NEWS [hunk 3]
```diff
@@ -1,3 +1,143 @@
...
+#1896
+#1898
+#1901
+#1904
+#1914
+#1926
+#1932
+#1953
+#1976
+#1978
```
Comment (peternewman) [inline NEWS:122]:
```suggestion
Add support for DMXControl Projects e.V. Nodle R4S interfaces
```

File: NEWS [hunk 4]
```diff
@@ -1,3 +1,143 @@
+x/y/2026 ola-0.11.0
+ Uncategorised and unsummarised PRs:
+#1000
+#1008
```
Comment (peternewman) [inline NEWS:4]:
```suggestion
* Add Favicons and manifest files to the new web UI
```

File: NEWS [hunk 5]
```diff
@@ -1,3 +1,143 @@
...
+#1301 - #1290
+#1303
+#1304
+#1305
+#1307
+#1310
+#1312
+#1314
+#1319
+#1326
```
Comment (peternewman) [inline NEWS:55]:
```suggestion
 * Add a new specific OLA_ASSERT_DMX_EQUALS test
```

File: NEWS [hunk 6]
```diff
@@ -1,3 +1,143 @@
...
+#1343
+#1348
+#1350
+#1355
+#1358
+#1359
+#1369
+#1380
+#1389
+#1391
```
Comment (peternewman) [inline NEWS:68]:
```suggestion
 * Renamed StatusMessageIdToString to StatusCodeToString in the
   E1.33 E133StatusHelper code to avoid ambiguity
```

File: NEWS [hunk 7]
```diff
@@ -1,3 +1,143 @@
...
+#1405
+#1410
+#1415
+#1417
```
Comment (peternewman) [inline NEWS:72]:
```suggestion
 * Fix some minor typos
 ```

File: NEWS [hunk 8]
```diff
@@ -1,3 +1,143 @@
...
+#1209
+#1213
+#1215
+#1231
+#1237
+#1242
+#1260
+#1268
+#1269
+#1270
```
Comment (peternewman) [inline NEWS:34]:
```suggestion
* Add an ola_trigger config to control relays using crelay
```

File: NEWS [hunk 9]
```diff
@@ -1,3 +1,143 @@
+x/y/2026 ola-0.11.0
+ Uncategorised and unsummarised PRs:
+#1000
```
Comment (aroffringa) [inline NEWS:3]:
```suggestion
* Initial Pychecker for OLA work
```

File: NEWS [hunk 10]
```diff
@@ -1,3 +1,143 @@
...
+#1080 - #1028
+#1100
+#1134
+#1136
+#1138
+#1140
+#1142
+#1146
+#1147
+#1176
```
Comment (aroffringa) [inline NEWS:21]: * Update some PLASA references to ESTA
* Adds support for AVLdiy.cn D512 usb dmx device, closes #1073
* Added DMXCreator 512 Basic USB interface
* Allow keyboard input for keypad
* More doxygen updates
* Markdown for plugin descriptions
* Several small Web UI improvements
* Support ShowJockey devices
* Standardise some test names

File: NEWS [hunk 11]
```diff
@@ -1,3 +1,143 @@
...
+#1196
+#1209
+#1213
+#1215
+#1231
+#1237
+#1242
+#1260
+#1268
+#1269
```
Comment (aroffringa) [inline NEWS:33]:
```suggestion
* Add some more RDM tests
* migrate all plugin descriptions to Markdown
* Add more missing RDM tests
* Fix an error that snuck in with #1185
* Add my RDM test listing/generation script
* More RDM tests
* Add java/OlaClient.getUniverseList()
* Display manufacturer name from the PID files to make life easier during RDM testing
* Fix configure to mangle AM_LDFLAGS instead of LDFLAGS, which it has no business to mangle.
* Update the link to the example plugin walkthrough
* Plugfest RDM changes
* More minor code tweaks
```

File: NEWS [hunk 12]
```diff
@@ -1,3 +1,143 @@
...
+#1275
+#1277
+#1289
+#1292
+#1294
+#1295
+#1297
+#1298
+#1299
+#1300
```
Comment (aroffringa) [inline NEWS:45]:
```suggestion
* More minor code fixes
* Added sensor ranges and units to RDM ModelCollector
* More RDM test changes
* SPI DMX receiver
* Support for Eurolite USB-DMX512-PRO MK2
* Yet More RDM tests
* More missing RDM tests
* Even more missing RDM tests
* Still even more missing RDM tests
* More IP RDM tests and tidy zero based ones
* Still even more RDM tests
```

File: NEWS [hunk 13]
```diff
@@ -1,3 +1,143 @@
...
+ Uncategorised and unsummarised PRs:
+#1000
+#1008
+#1014
+#1015
+#1035
+#1039
+#1045
+#1052
+#1067
```
Comment (aroffringa) [inline NEWS:11]:
```suggestion
*  Fix new flake8 issues due to latest pep8
*  OLA Trigger tidying and improvements
*  Grunt cleanup
*  Move common device info to device description, rather than ports
*  Fix typo in plugins/spi/SPIOutput.cpp
*  Fix a Python include.
*  Plugfest work and various RDM test fixes and improvements
```

File: NEWS [hunk 14]
```diff
@@ -1,3 +1,143 @@
...
+#1300
+#1301 - #1290
+#1303
+#1304
+#1305
+#1307
+#1310
+#1312
+#1314
+#1319
```
Comment (aroffringa) [inline NEWS:54]:
```suggestion
* clarfied/expanded build instructions for debian and similar distribut…
* More RDM test work
* Fix some more RDM tests
* Add some more missing tests, rename some others for clarity
* Still more RDM tests
* Still further RDM tests
* Various minor code tidying
* More Doxygen
```

File: NEWS [hunk 15]
```diff
@@ -1,3 +1,143 @@
...
+#1342
+#1343
+#1348
+#1350
+#1355
+#1358
+#1359
+#1369
+#1380
+#1389
```
Comment (aroffringa) [inline NEWS:67]:
```suggestion
* Initialize members of SPIDMXParser class
* Eliminate protobuf AddDescriptors() call.
* Ensure we populate the named UIDs in all circumstances
* Increment the transaction number for each RDM packet
* Initial version of Nanoleaf Aurora control
* Breaking API change, rename *syncron*(sic) to *synchron*
* Breaking API changes, spelling errors
* added APA102 with Pixel Brightness mode
* Fix a few more spelling issues
* Add link to my Bachelor's Thesis in SPI DMX Plugin's README
* Spi plugin port count
* Fix the Nanoleaf plugin ID I forgot to assign
```

File: NEWS [hunk 16]
```diff
@@ -1,3 +1,143 @@
...
+#1391
+#1405
+#1410
+#1415
```
Comment (aroffringa) [inline NEWS:71]:
```suggestion
* Rename ArtNet, Artnet, artnet, Art-net to Art-Net
* Feature/de interface
* Avoid use of root logger and fix close callback in python3
```

File: NEWS [hunk 17]
```diff
@@ -1,3 +1,143 @@
...
+#1417
+#1450
+#1457
+#1467
+#1476
+#1490
+#1494
+#1512
+#1520
```
Comment (aroffringa) [inline NEWS:80]:
```suggestion
* Detach kernel driver to allow claiming of USB interface
* Use Pandoc to render the GFM config file docs as HTML
* Extend configure so we switch to C++11 if necessary for Protobuf 3.6
* Add support for dimmer curves per E1.37-1
* Allow the README to header converter to handle double quotes
* Implement DIMMER_INFO, DIMMER_MINIMUM, and DIMMER_MAXIMUM PIDs per E1.37-1
* Add a screenshot mode which hides private data, also more RDM test and model collector improvements
* Fix some RDM discovery bugs
```

File: NEWS [hunk 18]
```diff
@@ -1,3 +1,143 @@
...
+#1522 - Towards #1396
+#1525
+#1532
+#1535
+#1556
+#1561
```
Comment (aroffringa) [inline NEWS:86]:
```suggestion
* Master resync and bump our POM minimum Java requirements
* Various tidying up/improvements
* Various improvements including fixing range validation
* Support for Eurolite MK2 with conflict checks
* Fix Getting Dimmer Max Value
```

File: NEWS [hunk 19]
```diff
@@ -1,3 +1,143 @@
...
+#1633
+#1647
+#1648
+#1683
+#1701
+#1705
+#1706
+#1712
+#1717
+#1720
```
Comment (aroffringa) [inline NEWS:100]:
```suggestion
* Protobuf 3.11 compatibility
* Explicitely link against tinfo. Needed for ncurses 6
* Allow all origin in HTTPServer responses (CORS)
* Add Stylelint CSS linter
* ola_rdm_get/set should only display actionable PIDs
* Add in/out point support to ola_recorder
* Extend examples/streaming-client to read universe numbers from STDIN
* Fix for old versions of libmicrohttpd and older compilers
* Fix make check on older compilers (ULL in UIDTest)
* Fix some more Python 3 errors
* usbdmx: Move widget deletion to main thread
* Set a default value for the enabling Eurolite Mk2 preference
* More thorough checking of SLOT_INFO and DEFAULT_SLOT_VALUE edge cases
```

File: NEWS [hunk 20]
```diff
@@ -1,3 +1,143 @@
...
+#1888
+#1896
+#1898
+#1901
+#1904
+#1914
+#1926
+#1932
+#1953
+#1976
```
Comment (aroffringa) [inline NEWS:121]:
```suggestion
* Ensure the dummy data we provide is binary to be compatible with Python 3
* Something else we support apparently  
* Add a helper method to convert an RDMResponse to a NACKing RDMResponse
* Pack and unpack UIDSets
* Update JUnit in pom.xml. Fixes https://github.com/OpenLightingProject/ola/security/dependabot/1
* Update manufacturer names
* More Python 3 RDM test compatibility
* Refresh the manufacturer names list and remove the duplicates too. Also master resync
* Fix codespell typo
* Update Debian Package Services and Config Directory
* Add a Eurolite serial number check
* Add some testing of MAC addresses within messaging fields
* Add the ability to set a universe to blackout instead via ola_set_dmx
* Initial IPv6 classes
* Do some minor formatting tweaks
* Fix spelling/grammar in RPC error message
* Add support for the IPv6 data type to the RDM messaging
* Track Eurolite MK2 by actual serial number where available
* Add initial E1.33 RDM tests
* Add support for the uint64 and int64 data types to the RDM messaging
```

File: NEWS [hunk 21]
```diff
@@ -1,3 +1,143 @@
...
+#1978
+#1981
+#1983
```
Comment (aroffringa) [inline NEWS:124]:
```suggestion
* Check if an interface is listed multiple times in LIST_INTERFACES
* Fix build for macOS where endian.h is not available
```

File: NEWS [hunk 22]
```diff
@@ -1,3 +1,143 @@
...
+#1986
+#1996
+#1999
+#2004
+#2014
+#2032
+#2041
```
Comment (aroffringa) [inline NEWS:131]:
```suggestion
* Add some E1.37-5 PIDs and accompanying RDM tests
* More RDM Responder tests
* E1.20 2025 Updates
* Tidy the Mac OS Endian behaviour
* E1.33 RDM Tests and other improvements
* Add more TSP slot ID definitions 
```

File: NEWS [hunk 23]
```diff
@@ -1,3 +1,143 @@
...
+#2042
+#2046
+#2052
+#2054
```
Comment (aroffringa) [inline NEWS:135]:
```suggestion
* Plugfest discovered fixes
* Modify compiler flags to suppress variable length array error
* RDM parameter info updates
```

**Comment by aroffringa** [comment]
I've added suggestions to replace all PR nrs by there title, except for the nrs for which you added a second nr -- not sure what you want wint those. Working in suggestions like this is a bit tedious I must say, much easier to directly edit the PR.

**Comment by aroffringa** [comment]
@peternewman shall we apply the suggested changes to the news file?

**Comment by aroffringa** [comment]
@peternewman maybe at this point it would be good to make the threshold for releasing a bit lower? We want that release! :) I would just apply the suggestions that are currently here, and then press the release button. I think master is good enough for a release.

File: NEWS (hunk 23 above)
Comment (peternewman) [reply NEWS]:
```suggestion
* Fix a bunch of minor typos
* Modify compiler flags to suppress variable length array error
* Update manufacturer name data to cover latest from ESTA allocations
```

File: NEWS (hunk 22 above)
Comment (peternewman) [reply NEWS]:
```suggestion
* Fix formatting of PID values in errors in RDM responder tests
* Update ESTA PIDs and data to cover E1.20-2025
* Add ESTA PIDs and data for E1.37-5
* Add some E1.37-5 RDM tests
* Flag potentially redundant RDM tests within our list_rdm_tests tool due to a PID's maximum data size already being at the max PDL
* Add some E1.33 RDM tests
* Handle a device info field being unavailable for some reason breaking the RDM model collector
* Add additional TSP slot ID definitions
```

File: NEWS (hunk 21 above)
Comment (peternewman) [reply NEWS]:
```suggestion
* Check if an interface is listed multiple times in the RDM test for LIST_INTERFACES
* Fix the build for macOS where endian.h is not available
```

File: NEWS [hunk 24]
```diff
@@ -1,3 +1,147 @@
...
+* Flag potentially redundant RDM tests within our list_rdm_tests tool due to a PID's maximum data size already being at the max PDL
+* Add some E1.33 RDM tests
+* Handle a device info field being unavailable for some reason breaking the RDM model collector
+* Add additional TSP slot ID definitions
+#2042
+* Fix a bunch of minor typos
+* Modify compiler flags to suppress variable length array error
+* Update manufacturer name data to cover latest from ESTA allocations
+
+
```
Comment (peternewman) [inline NEWS]:
```suggestion
* Collect the new E1.37-5 URLs and metadata via the RDM model collector
* Add support for a lot of the E1.37-5 PIDs to the RDM dummy responders
* Add additional acronyms to customise capitalization behaviour on
```

File: NEWS (hunk 18 above)
Comment (peternewman) [reply NEWS]: * Bump our POM minimum Java requirements
* Prettify the version option in the auto-generated flags man pages
* Switch more tests to OLA_ASSERT_EQ for better reporting on failures
* Ensure int descriptors are within the valid ranges and/or labels
* Deprecate a duplicate RDM Start Code constant, use ola::rdm::START_CODE from now on
* Fix a bug in the responder tests that happens in some scenarios with queued messages
* Add support for the Eurolite USB-DMX512 PRO MK2, disabled by default due to clashes with various other FTDI based interfaces
* Fix GetDimmerMaximumLevel in the RDMAPI only processing one of the two bytes of the value

**Comment by peternewman** [comment]
> I've added suggestions to replace all PR nrs by there title,

> @peternewman maybe at this point it would be good to make the threshold for releasing a bit lower?

Thanks, although, and I'm clearly not the only person guilty of it, given this wasn't me for a change, `Feature/de interface` but PR titles often hide many more useful notes, or are just a bit generic, so I think we need to finesse them a little.

Although I also accept I've got distracted by lots of other stuff and not worked enough on pushing this over the line recently.

> except for the nrs for which you added a second nr -- not sure what you want wint those.

The second number was the issue number, which we traditionally mention in the change log.

> Working in suggestions like this is a bit tedious I must say, much easier to directly edit the PR.

Noted for next time, I thought it might be easier to collaborate on refining them like this/not risk creating conflicts (plus I wasn't sure if you'd actually have edit rights on my PR). Although something has gone a bit wonky and it's now showing a delta of the entire file after merging one suggestion...

File: NEWS (hunk 16 above)
Comment (peternewman) [reply NEWS]: * Rename ArtNet, Artnet, artnet, Art-net to Art-Net #1328
* Add support for the Digital Enlightenment USB-DMX interface #241
* Add support for the FX5 DMX interface #241
* Avoid the use of the root logger in the Python API
* Fix closed socket detection in python3

File: NEWS (hunk 17 above)
Comment (peternewman) [reply NEWS]: * Detach kernel driver to allow claiming of USB interface when using libusb
* Use Pandoc to render the GFM config file docs as HTML on the website
* Extend configure so we switch to C++11 if necessary for Protobuf 3.6 #1463
* Add support for various dimmer parameter PIDs from E1.37-1 in the web UI and HTTP API
* Correctly handle double quotes in the README to header converter #1489
* Add a screenshot mode which hides private data
* Add additional acronyms to customise capitalization behaviour
* Collect the device label and additional basic personality info in the RDM model collector
* Unify the GetOutOfRange and SetOutOfRange Mixins in the RDM responder tests
* Fix some RDM discovery bugs around suspected muted responders continuing to respond
* Fix a bug in how we simulate RDM responses and collisions in our discovery test cases

File: NEWS (hunk 19 above)
Comment (peternewman) [reply NEWS]: * Allow all origin in HTTPServer responses (CORS)
* Add Stylelint CSS linter and fix the issues it found #1610
* ola_rdm_get/set should only display actionable PIDs #1632
* Add in/out point support to ola_recorder #1602
* Extend ola_streaming_client to optionally read universe numbers from STDIN
* Fix CORS support in older versions of libmicrohttpd
* Fix make check on older compilers
* Fix lots of Python 3 errors
* Move widget deletion to main thread in usbdmx plugin to avoid potential deadlock
* More thorough checking of SLOT_INFO and DEFAULT_SLOT_VALUE edge cases in both our responder helpers and the RDM responder tests

File: NEWS (hunk 15 above)
Comment (peternewman) [reply NEWS]: * Initialize members of SPIDMXParser class
* Remove Protobuf AddDescriptors() call to support newer Protobuf #1192
* Ensure we populate the named UIDs for the RDM responder tests in all circumstances
* Increment the transaction number for each RDM packet #189
* Add support for Nanoleaf Aurora control
* Renamed *syncron*(sic) to *synchron* across all our code #1331
* Renamed SC_E133_NONEXISTANT_ENDPOINT(sic) to SC_E133_NONEXISTENT_ENDPOINT in the E133Enums
* Renamed sucessful(sic) to successful in various parts of usbpro_firmware
* Add support for APA102 with pixel brightness mode #1356
* Increase SPI plugin's max port count to 32 #1379

File: NEWS (hunk 13 above)
Comment (peternewman) [reply NEWS]: * Improve our flake8 results and cleanliness
* Tidy some ola_trigger code and Doxygen documentation
* Support unquoted numeric variables, arguments and assignment within ola_trigger
* Fix double counted line numbers in ola_trigger configs
* Cleanup the Grunt config on the new web UI
* Move common device info to device description, rather than ports for USBPro and SPI
* Fix a typo in the SPI output length calculation
* Fix a Python include path
* Add support for USB Pro MK II B widgets
* Make the RDM responder test GetMaxPacketSize test data increment uniquely to help with diagnostics
* Add more unsupported SET tests to the RDM responder tests
* Fix the MuteAllDevices RDM responder test for plugins that don't support discovery commands
* Actually decode PID param data and correct the flag logic in the logic_rdm_sniffer

File: NEWS (hunk 14 above)
Comment (peternewman) [reply NEWS]: * Clarified/expanded build instructions for Debian and similar distributions
* Ensure we always stop the RDM responder tests when we've finished testing, so they complete correctly
* Standardise more RDM responder test names based on their PID names
* Add some dimmer SET  error case tests to the RDM responder tests
* Ensure the RDM model collector can continue after an unpack error for a single PID
* Add additional Status Report Threshold and Status Messages RDM responder tests
* Add extra E1.37-2 error case RDM responder tests
* Improve more of our Doxygen documentation

File: NEWS (hunk 12 above)
Comment (peternewman) [reply NEWS]: * Confirm existing support for LumenRadio CRMX Nova TX USB
* Tidy RDM Command processing
* Simplify dummy responder config code
* Added sensor ranges and units to RDM Model Collector
* Improve some RDM responder test DMX Personality Description tests
* Add native SPI DMX support to receive DMX input using an SPI port samping at 2MHz #1285
* Add support for Eurolite USB-DMX512-PRO MK2
* Add Status ID Description and Parameter Description without data RDM responder tests
* Add missing no data, with data and with extra data RDM responder tests
* Add E1.37-2 based RDM responder tests
* Simplify the Get and Set Zero Mixins for the RDM responder tests

File: NEWS (hunk 20 above)
Comment (peternewman) [reply NEWS]: * Ensure the dummy data we provide in our RDM responder tests is binary for Python 3 compatibility
* Document apparent existing support for the Eurolite freeDMX Wi-Fi (from it's manual)
* Add a helper method to convert an RDMResponse to a NACKing RDMResponse
* Add the ability to Pack and unpack UIDSets
* Update JUnit in our POM
* Update RDM manufacturer names and remove duplicates #1837
* Fix a number of typos and grammar issues, primarily caught by codespell
* Update Debian Package Services and Config Directory including adding SystemD support and a build name which is also shown in various places
* Add Eurolite serial number support, so specific FTDI interfaces can be claimed as Eurolite based on their serial numbers
* Add testing of MAC addresses within our RDM messaging
* Add the ability to set a universe to blackout via ola_set_dmx
* Add some initial IPv6 classes for internal use and via the RDM messaging
* Add additional E1.33 and E1.37-7 PIDs
* Track Eurolite MK2 by serial number where available so patching should follow the device
* Add some E1.33 RDM responder tests
* Add support for uint64 and int64 internally and via the RDM messaging

File: NEWS (hunk 11 above)
Comment (peternewman) [reply NEWS]: * Migrate plugin descriptions to Markdown
* Add a script to list expected RDM responder tests and autogenerate test code for them
* Renamed *MiniumLevel*(sic) to *MinimumLevel* in various RDM responder test names
* Add the ability to get info about all universes to the Java API
* Display manufacturer name from the PID files during RDM responder testing
* Configure now correctly mangles AM_LDFLAGS instead of LDFLAGS #1201
* Fix the path to the example OSC plugin walkthrough
* Fix the categorisation of some RDM responder tests
* Simplify some code by using our ToHex formatting code more

File: NEWS (hunk 9 above)
Comment (peternewman) [reply NEWS]: * Improve our pychecker results and cleanliness

**Comment by peternewman** [comment]
Closing this to be superseded by #2076 due to a wonky bulk apply of suggestions which ended up changing the line endings of the whole file.
